### DISCUSSION TITLE
need details about `shadowhook_hook_func_addr`

### DISCUSSION THREAD

**Discussion by MhmRdd** [bytedance/android-inline-hook#84] [Q&A] [answered]
I successfully hooked some functions with `shadowhook_hook_func_addr`.
the question here is: does shadowhook recall the original function after the hooked function finishes or only when specified inside the hooked function?

`int (*a)(unsigned char* a1, unsigned char* a2, int a3);
int b(unsigned char* a1, unsigned char* a2, int a3)
{
    // do something
    return 0;
}
stub = shadowhook_hook_func_addr((void*) (someexec), a, (void**) &b);`

above is a sample code of hooking, if anyone can explain what's going to happen?

**Comment by Colibrow** [comment, marked as answer]
This question is actually addressed in our documentation. [shadowhook_hook_func_addr](https://github.com/bytedance/android-inline-hook/blob/main/doc/manual.md#2-hook-function-without-symbol-info-via-function-address).
Whether to call the original function depends on the initialized mode". [here](https://github.com/bytedance/android-inline-hook/blob/main/doc/manual.md#proxy-functions-in-shared-mode).
